### PR TITLE
MINOR: ownerName missing field after https://github.com/open-metadata/OpenMetadata/pull/19423

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes:

Fixes #<issue-number>

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### High-level design:

N/A — small change. 

#
#### Manual testing performed

#
### UI screen recording / screenshots:

Not applicable. 

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] My PR is linked to a GitHub issue via `Fixes #<issue-number>` above.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.
- [ ] For UI changes: I attached a screen recording and/or screenshots above.
- [ ] I have added tests (unit / integration / Playwright as applicable) and listed them above.

----
## Summary by Gitar

- **Data insights enrichment:**
  - Added `OwnerNameStep` to project flat `ownerName` list onto data insight snapshots.
  - Registered `OwnerNameStep` in `DataInsightsEntityEnricherProcessor` and configured it in `config.json`.
- **Testing:**
  - Added unit tests in `DataInsightsEntityEnricherProcessorTest` covering owned, unowned, and missing owner name scenarios.

<sub>This will update automatically on new commits.</sub>

<details open><summary><h3>Greptile Summary</h3></summary>

This PR restores the `ownerName` flat keyword field that was dropped from Data Insights snapshots after PR #19423 refactored owner handling. A new `OwnerNameStep` is introduced that mirrors what `SearchIndex.populateCommonFields` does, projecting each owner's name into a flat list so `query_string` DI filters (e.g. `ownerName: *`) can reach it without traversing the nested `owners` array.

- **`OwnerNameStep`**: reads `entity.getOwners()`, filters out null names, and writes the result under the `ownerName` key — absent entirely when no owners or all names are null, so existence filters correctly exclude unowned assets.
- **Pipeline wiring**: `OwnerNameStep` is inserted after `OwnerTeamStep` in `DataInsightsEntityEnricherProcessor`, and `ownerName` is added to the `common` mapping fields in `config.json`.
- **Tests**: three focused unit tests cover the owned, unowned, and all-null-name edge cases with a lightweight `MockOwnedEntity` helper.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the change is a targeted restoration of a missing field with no side effects on existing enrichment steps.

All four files are small and focused. OwnerNameStep is purely additive, guards null/empty owners and null individual names, and the pipeline contract (additive-only, no cross-step reads) is respected. Three unit tests cover the key behavioral cases. The config.json addition is consistent with the existing field list. No data loss or behavioural regression is possible — absent ownerName was the prior broken state.

No files require special attention.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/enricher/steps/OwnerNameStep.java | New enrichment step that projects a flat ownerName keyword list from the nested owners array. Correctly guards against null/empty owners and null individual names. |
| openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/DataInsightsEntityEnricherProcessor.java | Wires OwnerNameStep into the enrichment pipeline after OwnerTeamStep, restoring the missing field. |
| openmetadata-service/src/main/resources/dataInsights/config.json | Adds ownerName to the common mappingFields list so the projected field is included in data insights documents. |
| openmetadata-service/src/test/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/DataInsightsEntityEnricherProcessorTest.java | Adds three unit tests covering the owned, unowned, and all-null-name cases for OwnerNameStep, plus a supporting MockOwnedEntity helper class. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[EnrichmentPipeline] --> B[IdentityProjectionStep]
    B --> C[DescriptionSourcesStep]
    C --> D[TagAndTierSourcesStep]
    D --> E[OwnerTeamStep - writes 'team']
    E --> F[OwnerNameStep - writes 'ownerName' NEW]
    F --> G[TierStep]
    G --> H[DescriptionStatsStep]
    H --> I[CustomPropertiesStep]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[EnrichmentPipeline] --> B[IdentityProjectionStep]
    B --> C[DescriptionSourcesStep]
    C --> D[TagAndTierSourcesStep]
    D --> E[OwnerTeamStep - writes 'team']
    E --> F[OwnerNameStep - writes 'ownerName' NEW]
    F --> G[TierStep]
    G --> H[DescriptionStatsStep]
    H --> I[CustomPropertiesStep]
```

</a>
</details>

<sub>Reviews (1): Last reviewed commit: ["fix: ownerName missing field after https..."](https://github.com/open-metadata/openmetadata/commit/a2bd7c235fa488700789bdb2bae8e63a808e7a93) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=43060695)</sub>

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/enricher/steps/OwnerNameStep.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public void apply(EnrichmentTarget target) {
+    List<EntityReference> owners = target.entity().getOwners();
+    if (nullOrEmpty(owners)) {
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/enricher/steps/OwnerNameStep.java:40]: nit: if (!nullOrEmpty(owners))

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/enricher/steps/OwnerNameStep.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      return;
+    }
+    List<String> ownerNames =
+        owners.stream().map(EntityReference::getName).filter(name -> !nullOrEmpty(name)).toList();
+    if (!ownerNames.isEmpty()) {
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/enricher/steps/OwnerNameStep.java:45]: will this be empty?

Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/dataAssets/processors/enricher/steps/OwnerNameStep.java:45]:
```
List<String> ownerNames =
        owners.stream().map(EntityReference::getName).filter(name -> !nullOrEmpty(name)).toList();
```

It would be an edge case where all the name for the owner are null or empty, then we would return an empty list

**Review by harshach — APPROVED** [review]
